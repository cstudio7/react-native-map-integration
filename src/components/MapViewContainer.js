import React, {Component} from 'react';
import {StyleSheet, Text, View, ActivityIndicator} from 'react-native';
import MapView, {Marker} from 'react-native-maps';

import styles from "./../styles/styles";

const defaultProps = {
    lat: "" ,
    lng: "",
    title: ""
}
export default class MapViewContainer extends Component<Props> {
  render() {
    const {lat, lng, title} = this.props;
    return (
      <View style ={styles.container}>
        { (lat !== "" && lng !== "") ?
          <MapView
            style={styles.map}
            region={{
              latitude: lat,
              longitude: lng,
              latitudeDelta: 0.0900,
              longitudeDelta: 0.0500,
            }}>
            <Marker
              coordinate={{
                  latitude: lat,
                  longitude: lng
              }}
              title={title}/>
          </MapView>
          :
          <View style={[styles.container, styles.horizontal]}>
            <ActivityIndicator size="large" color="#0000ff" />
          </View>
        }
     </View>
    );
  }
}

MapViewContainer.defaultProps = defaultProps;
