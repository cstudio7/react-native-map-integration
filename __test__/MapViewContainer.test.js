import React from "react";
import renderer from "react-test-renderer";
import MapViewContainer from "../src/components/MapViewContainer";

describe("Render MapViewContainer component", () => {
    const wrapper = renderer.create(<MapViewContainer />);
    it("should render component", () => {
        const wrapperInstance = wrapper.toJSON();
        expect(wrapperInstance).toMatchSnapshot();
    });
});
