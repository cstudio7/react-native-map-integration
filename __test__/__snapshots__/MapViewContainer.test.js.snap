// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MapViewContainer component should render component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "padding": 10,
        },
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#0000ff"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;
